hera_pspec/pspecdata.py

```diff
@@ -2401,6 +2693,11 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None,
             given an input visibility noise model, and store the output
             in the UVPSpec object.
 
+        save_q : bool, optional
```

Comment:
I think it would be more economical to add a store_M option and store the M-matrices in the uvpspec object so that users can recover q by applying and inverted M to data_array.

Reply:
Well, I think that the idea here is that q_hat is already computed en-route to getting p_hat, so if by economical you mean the least amount of computation, then I think the answer is to return the pre-computed q_hat if requested (default should be False b/c this is a non-standard situation I think).

Reply:
If we are going to keep this, @JianrongTan let's change the kwarg to `return_q`

Reply:
I think keeping `return_q` to False would make me happy.

Reply:
So do you mean `return_q` refers to store the unnormalized one other than the normalized one? So we don't have double outputs for `uvp` and `uvp_q`, instead, we have output for either `uvp` or `uvp_q`?

Reply:
Yeah that could be confusing--here's a question, do you ever find yourself needing both p_hat and q_hat? If not, then you could make `return_q=False` just return uvp and `return_q=True` just return uvp_q. Otherwise you can make it `return uvp, uvp_q` and if `return_q=False` set `uvp_q=None`